pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -1188,40 +1210,93 @@ def network_init(self):
             cmd.append(str(n['protocol_version']))
 
         self.run_neard(cmd)
-        self.set_state(TestState.SET_VALIDATORS)
-        self.save_data()
+        self.set_state(TestState.SETTING_UP_PROTOCOL)
+
+    def _setup_bucket_path(self):
+        mocknet_id = self.config['mocknet_id']
+        run_id = self.run_id()
+        return f"gs://near-mocknet-artefact-store/{mocknet_id}/{run_id}"
+
+    def check_setup_bucket(self):
+        """Check if setup bucket exists and download the contents into the
+        target home folder if so."""
 
-    def check_set_validators(self):
+        # Check if setup bucket exists.
+        check_cmd = ['gsutil', 'ls', self._setup_bucket_path()]
+        logging.info(f"Checking if setup bucket exists: {' '.join(check_cmd)}")
+
+        result = subprocess.run(check_cmd, capture_output=True, text=True)
+        if result.returncode == 1:
+            logging.info(f"Setup bucket not found")
+            return
+        elif result.returncode != 0:
+            logging.error(f"Error checking setup bucket: {result.stderr}")
+            self.set_state(TestState.ERROR)
```

Comment:
Maybe set error with message. This will show up in the state of he mocknet and make it easier to debug.
```suggestion
            error_msg = f"Error checking setup bucket: {result.stderr}"
            logging.error(error_msg)
            self.set_state(TestState.ERROR, data=error_msg)
```